crates/worker/src/partition/state_machine/command_interpreter/mod.rs

```diff
@@ -1358,6 +1362,189 @@ where
                     );
                 }
             }
+            EnrichedEntryHeader::GetPromise { is_completed, .. } => {
+                if !is_completed {
+                    let_assert!(
+                        Entry::GetPromise(GetPromiseEntry { key, .. }) =
+                            journal_entry.deserialize_entry_ref::<Codec>()?
+                    );
+
+                    if let Some(service_id) =
+                        invocation_metadata.invocation_target.as_keyed_service_id()
```

Comment:
Do I understand it correctly that the promise feature only allows a service instance to access promises that were created by itself? I guess if we wanted to do it generally, one would have to introduce a new message that can retrieve the result from a different partition.

Reply:
> Do I understand it correctly that the promise feature only allows a service instance to access promises that were created by itself?

Yes this is how it's modeled in the service protocol entry message too (it doesn't have the service name/service key of the promise holder). The idea is that people should not expose promises to foreign services as api surface, but should rely on shared handlers for that.